src/sweeps/bayes_search.py

```diff
@@ -444,11 +457,15 @@ def _construct_gp_data(
     if len(y) > 0:
         y[~np.isfinite(y)] = worst_metric
 
+    warnings = ""
+    if len(y) == 0:
+        warnings += "\nSweep has no valid samples of the metric."
```

Comment:
this condition does not capture all of the cases, see line 458 above. there can still be invalid values of the metric even if `len(y) > 0`. but by this point we have imputed them with `worst_metric`. we should move this block before line 457 but after line 456, and change it to 
```suggestion
    if y[~np.asarray(list(map(is_number, y)))].size == len(y):
        warnings += "\nSweep has no valid samples of the metric."
```

where `is_number` is defined in `run.py`. we may have to move that over from `run.py` to a utility file